bpf/lib/proxy.h

```diff
@@ -44,6 +187,8 @@ __ctx_redirect_to_proxy(struct __ctx_buff *ctx, void *tuple __maybe_unused,
 			__be16 proxy_port, bool from_host __maybe_unused,
 			bool ipv4 __maybe_unused)
 {
+	int result = CTX_ACT_OK;
+
 	ctx->mark = MARK_MAGIC_TO_PROXY | proxy_port << 16;
```

Comment:
Yes. The `MARK_MAGIC_TO_PROXY` is still required for routing the packets locally, even if we assign the socket in this function at TC ingress.

In the case where this function is called at a TC egress hook point, it is additionally required to pass the port along to the subsequent eBPF program at TC ingress which will handle the socket lookup.

Reply:
I see. And `proxy_port` will always be zero in TC ingress case, no? Would it be worth writing the proxy port in the mark only in the TC egress case? I mostly want to avoid trashing mark bits if we don't need to.

Reply:
I think it's a little more subtle than how you put it, but yes I believe that if we're at TC ingress we don't need to set the bits that we would otherwise dedicate to the proxy port. I'll look into another patch for this.

Reply:
Proposed incremental:

```
diff --git a/bpf/lib/proxy.h b/bpf/lib/proxy.h
index 1efdc534ec83..cf7157bbc7d4 100644
--- a/bpf/lib/proxy.h
+++ b/bpf/lib/proxy.h
@@ -194,7 +194,12 @@ __ctx_redirect_to_proxy(struct __ctx_buff *ctx, void *tuple __maybe_unused,
 {
        int result = CTX_ACT_OK;
 
-       ctx->mark = MARK_MAGIC_TO_PROXY | proxy_port << 16;
+#ifdef ENABLE_TPROXY
+       if (!from_host)
+               ctx->mark |= MARK_MAGIC_TO_PROXY;
+       else
+#endif
+               ctx->mark = MARK_MAGIC_TO_PROXY | proxy_port << 16;
 
 #ifdef HOST_REDIRECT_TO_INGRESS
        cilium_dbg(ctx, DBG_CAPTURE_PROXY_PRE, proxy_port, 0);
```